src/solve/moi.jl

```diff
@@ -47,126 +123,99 @@ function MOI.eval_hessian_lagrangian(moiproblem::MOIOptimizationProblem{T}, h, x
     return
 end
 
-function MOI.eval_constraint_jacobian(moiproblem::MOIOptimizationProblem, j, x)
-    isempty(j) && return
-    moiproblem.f.cons_j === nothing && error("Use OptimizationFunction to pass the derivatives or automatically generate them with one of the autodiff backends")
-    n = length(moiproblem.u0)
-    moiproblem.f.cons_j(moiproblem.J, x)
-    for i in eachindex(j)
-        j[i] = moiproblem.J[i]
-    end
-end
-
-function MOI.jacobian_structure(moiproblem::MOIOptimizationProblem)
-    return Tuple{Int,Int}[(con, var) for con in 1:size(moiproblem.J,1) for var in 1:size(moiproblem.J,2)]
```

Comment:
The two for-loops here were back-to-front, which meant the Jacobian passed to Ipopt was incorrect. @ccoffrin this was the cause of your problem.